src/components/engagement/engagement.service.ts

```diff
@@ -607,7 +613,15 @@ export class EngagementService {
         ? e.withField('engagement.projectId')
         : e;
     }
+
     if (project.status !== ProjectStatus.InDevelopment) {
+      const changeRequests = await this.projectService.listChangeRequests(
+        project,
+        ProjectChangeRequestListInput.defaultVal,
+        session
+      );
+      if (project.status === ProjectStatus.Active && changeRequests.total > 0)
+        return;
```

Comment:
I don't think this is necessary, unless I'm missing something. Having a `changeset` passed in means this change is for a changeset, which is where we would skip this check. Just because a project has a pending change-request/changeset does not mean an Engagement can be created on the Project _outside_ that changeset.

Reply:
Also the business logic is duplicated [here](https://github.com/SeedCompany/cord-api-v3/blob/5275777d82f773be0f60000c0de39dba36cd02a9/src/components/project/project.service.ts#L452-L455) sadly, and needs to be updated as well.

Reply:
I must've misunderstood what you were telling me then, because taking out the changeRequest logic fails a bunch of tests in the engagement-changeset-aware suite, and that's why I put that stuff in in the first place.

Reply:
You said "Yeah that needs to be updated to ignore if changing inside of a changeset". The stuff to be ignored being: if a project is any other status other than "Active", and the user adding the engagement is not an Admin, error out.

Reply:
Just to be clear, the [if statement change above this](https://github.com/SeedCompany/cord-api-v3/pull/2400/files/332d87a56b6bf41228ee76c41fc7600d34aca14d..8acee685afac3880c31a49310ceaf413eb651670#diff-64ecdc9d362652a540d48b634cbc48b1758180b1836723f5b3926646b04452eaR604) (outside of this diff) is correct. Does that change your comment?